File: Angular/src/app/invoice/interfaces/invoice-detail.ts
export class InvoiceDetail {
  id: null | undefined;
  invoiceId: number;
  codUnidadMedida: string;
  ctdUnidadItem: number;
  codProducto: string;
  codProductoSunat: string;
  desItem: string;
  mtoValorUnitario: number;
  sumTotTributosItem: number;
  codTriIgv: string;
  mtoIgvItem: number;
  mtoBaseIgvItem: number;
  nomTributoIgvItem: string;
  codTipTributoIgvItem: string;
  tipAfeIgv: string;
  porIgvItem: string;
  codTriIcbper: string;
  mtoTriIcbperItem: number;
  ctdBolsasTriIcbperItem: number;
  nomTributoIcbperItem: string;
  codTipTributoIcbperItem: string;
  mtoTriIcbperUnidad: number;
  mtoPrecioVentaUnitario: number;
  mtoValorVentaItem: number;
  mtoValorReferencialUnitario: number;

  constructor() {
    this.id = null;
    this.invoiceId = 0;
    this.codUnidadMedida = '';
    this.ctdUnidadItem = 0;
    this.codProducto = '';
    this.codProductoSunat = '';
    this.desItem = '';
    this.mtoValorUnitario = 0;
    this.sumTotTributosItem = 0;
    this.codTriIgv = '';
    this.mtoIgvItem = 0;
    this.mtoBaseIgvItem = 0;
    this.nomTributoIgvItem = '';
    this.codTipTributoIgvItem = '';
    this.tipAfeIgv = '';
    this.porIgvItem = '';
    this.codTriIcbper = '';
    this.mtoTriIcbperItem = 0;
    this.ctdBolsasTriIcbperItem = 0;
    this.nomTributoIcbperItem = '';
    this.codTipTributoIcbperItem = '';
    this.mtoTriIcbperUnidad = 0;
    this.mtoPrecioVentaUnitario = 0;
    this.mtoValorVentaItem = 0;
    this.mtoValorReferencialUnitario = 0;
  }
}

File: Database/Services/Facturador/InvoiceDetail.cs
namespace Nebula.Database.Services.Facturador;

public class InvoiceDetail
{
    public string codUnidadMedida { get; set; } = string.Empty;
    public string ctdUnidadItem { get; set; } = string.Empty;
    public string? codProducto { get; set; } = null;
    public string? codProductoSUNAT { get; set; } = null;
    public string desItem { get; set; } = string.Empty;
    public string mtoValorUnitario { get; set; } = string.Empty;
    public string sumTotTributosItem { get; set; } = string.Empty;
    #region Tributo: IGV(1000)
    public string codTriIGV { get; set; } = "1000";
    public string mtoIgvItem { get; set; } = "0.00";
    public string mtoBaseIgvItem { get; set; } = "0.00";
    public string nomTributoIgvItem { get; set; } = "IGV";
    public string codTipTributoIgvItem { get; set; } = "VAT";
    public string tipAfeIGV { get; set; } = string.Empty;
    public string porIgvItem { get; set; } = string.Empty;
    #endregion
    public string codTriISC { get; set; } = "-";
    public string codTriOtro { get; set; } = "-";
    #region Tributo ICBPER 7152
    public string codTriIcbper { get; set; } = "-";
    public string mtoTriIcbperItem { get; set; } = "0.00";
    public string ctdBolsasTriIcbperItem { get; set; } = "0";
    public string nomTributoIcbperItem { get; set; } = "ICBPER";
    public string codTipTributoIcbperItem { get; set; } = "OTH";
    public string mtoTriIcbperUnidad { get; set; } = "0.00";
    #endregion
    public string mtoPrecioVentaUnitario { get; set; } = string.Empty;
    public string mtoValorVentaItem { get; set; } = string.Empty;
    public string mtoValorReferencialUnitario { get; set; } = "0.00";
}
